zipkin/src/test/java/zipkin/storage/SpanStoreTest.java

```diff
@@ -205,9 +205,10 @@ public void getSpanNames_allReturned() {
   }
 
   @Test
-  public void getAllServiceNames() {
+  public void getAllServiceNames_mergesAnnotation_andBinaryAnnotation() {
+    // creates a span with mutual exclusive endpoints in binary annotations and annotations
     BinaryAnnotation yak = BinaryAnnotation.address("sa", Endpoint.create("yak", 127 << 24 | 1));
-    accept(span1.toBuilder().addBinaryAnnotation(yak).build(), span4);
+    accept(span1.toBuilder().binaryAnnotations(asList(yak)).build());
```

Comment:
before, a binary annotation had the same service name as a regular annotation, masking the bug where we weren't querying the latter